src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -440,11 +440,14 @@ private void AddLexicalClassifications(IClassificationService classificationServ
 
             using var _ = Classifier.GetPooledList(out var tempList);
 
-            // If we have a syntax root ready, use the direct, non-async/non-blocking approach to getting classifications.
-            if (!lastProcessedDocumentOrRoot.TryGetFirst(out var root))
-                classificationService.AddSyntacticClassificationsAsync(lastProcessedDocumentOrRoot.Second, span.Span.ToTextSpan(), tempList, cancellationToken).Wait(cancellationToken);
-            else
-                classificationService.AddSyntacticClassifications(solutionServices, root, span.Span.ToTextSpan(), tempList, cancellationToken);
+            // If we have a syntax root ready, use it.  Otherwise, get the root synchronously.  We do not want to do
+            // anything async here as we'd have to block on that, potentially starving the UI thread while the
+            // threadpool does work.
+            var root = lastProcessedDocumentOrRoot.TryGetFirst(out var tempRoot)
+                ? tempRoot
+                : lastProcessedDocumentOrRoot.Second.GetSyntaxRootSynchronously(cancellationToken);
+
+            classificationService.AddSyntacticClassifications(solutionServices, root, span.Span.ToTextSpan(), tempList, cancellationToken);
```

Comment:
@RikkiGibson Todd and I think we've found an NRT bug here.  'root' has the type `SyntaxNode?`.  However, we get no warnings about passing that type into AddSyntacticClassifications (which requires a `SyntaxNode`).  *Neither* this file, nor the interface where AddSyntacticClassifications is contained are nullable disabled.  So we have *no* idea why this isn't warning.  Seems *bad*.

Reply:
oh weird. it's fine in netstandard2.0 build, but not fine in net6.

Reply:
going ahead and approving even with the odd nullability issue

Reply:
> oh weird. it's fine in netstandard2.0 build, but not fine in net6.

This tracks for me. IIRC we try to only issue nullable warnings when targeting netcore, because the netstandard build is going to be using unannotated framework APIs. That would result in spurious warnings. If you think there is a risk in Roslyn of projects that don't target netcore at all missing nullable warnings, feel free to file a bug for it.